### PR TITLE
fix(app): Fix dropdown menu display issue for many items

### PR DESCRIPTION
**Pull request by koji**
# Overview
Fix dropdown menu display issue for many items
The current target is edge because this is UI issue. (talked about this with Shlok and Seth)

Mel updated the details
https://www.figma.com/design/l7BAJAGICr10ELsSqQD0Sr/Design-System%3A-Desktop?node-id=13031-89381&t=L2chZxRXwafZwhv0-0

RQA-2808

# Test Plan
1. test the following protocol
[Flex_S_v2_18_NO_PIPETTES_GoldenRTP.py.zip](https://github.com/user-attachments/files/15832059/Flex_S_v2_18_NO_PIPETTES_GoldenRTP.py.zip)

2. test the protocol Anurag shared
https://opentrons.slack.com/archives/CSCLVUW3C/p1718301002191389?thread_ts=1718298969.028039&cid=CSCLVUW3C

# Changelog

# Review requests

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] update storybook

File: app/src/atoms/MenuList/DropdownMenu.tsx [hunk 1]
```diff
@@ -213,9 +216,12 @@ export function DropdownMenu(props: DropdownMenuProps): JSX.Element {
             width={width}
             top={dropdownPosition === 'bottom' ? '2.5rem' : undefined}
             bottom={dropdownPosition === 'top' ? '2.5rem' : undefined}
+            overflowY={OVERFLOW_AUTO}
+            maxHeight="20rem" // Set the maximum display number to 10.
           >
             {filterOptions.map((option, index) => (
               <MenuItem
+                zIndex="3"
```
Comment (jerader) [inline app/src/atoms/MenuList/DropdownMenu.tsx:237]: does this work? i thought zindexes are numbers

```suggestion
                zIndex={3}
```

File: app/src/atoms/MenuList/DropdownMenu.tsx [hunk 2]
```diff
@@ -88,8 +95,9 @@ export function DropdownMenu(props: DropdownMenuProps): JSX.Element {
           scrollOffset = parentRect.top
         }
 
-        const dropdownBottom =
-          dropdownRect.bottom + (filterOptions.length + 1) * 34 - scrollOffset
+        const downSpace =
+          filterOptions.length + 1 > 10 ? 316 : (filterOptions.length + 1) * 34
```
Comment (jerader) [inline app/src/atoms/MenuList/DropdownMenu.tsx]: maybe we can save this 316 as a constant explaining what it is

File: app/src/atoms/MenuList/DropdownMenu.tsx (hunk 1 above)
Comment (koji) [reply app/src/atoms/MenuList/DropdownMenu.tsx:237]: on my end I use `{3}` lol

**Review by ncdiehl11 — COMMENTED** [review]
Code looks good with Jethary's changes and the long dropdown is scrollable as expected, but one of the non-scrollable dropdowns got cut off by the top of the slideout
Screenshot 2024-06-14 at 11 11 04 AM

**Comment by koji** [comment]
> Code looks good with Jethary's changes and the long dropdown is scrollable as expected, but one of the non-scrollable dropdowns got cut off by the top of the slideout Screenshot 2024-06-14 at 11 11 04 AM

is that Anurag's protocol?

**Comment by ncdiehl11** [comment]
> > Code looks good with Jethary's changes and the long dropdown is scrollable as expected, but one of the non-scrollable dropdowns got cut off by the top of the slideout Screenshot 2024-06-14 at 11 11 04 AM
> 
> is that Anurag's protocol?

yes— the 'starting tip' parameter

**Review by jerader — APPROVED** [review]
🚗

**Comment by koji** [comment]
ToDo
- [ ] try to use `react-select` for DropdownMenu after finish feature work

**Pull request merged**